sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c

```diff
@@ -56,18 +56,6 @@
 #define CLIENT_SECRET_ENV "AZURE_CLIENT_SECRET"
 #define URI_ENV "AZURE_KEYVAULT_URL"
 
-#define TEST_FAIL_ON_ERROR(result, message) \
-  do \
-  { \
-    if (az_failed(result)) \
-    { \
-      printf("\n"); \
-      printf(message); \
-      printf("\n"); \
-      return 1; \
-    } \
-  } while (0)
```

Comment:
I see a lot of use of having this MACRO.
Formatting error message for all places, and also easy way to introduce a clean up routine. (will be needed for libcurl)

I think explicitly adding all the `if () else () ` statements makes the sample grow bigger and distract customer from the main idea of the sample  by looking all of that duplicated code